src/definition/populate_test.go

```diff
@@ -456,75 +421,97 @@ func TestIntegrationProtocol2PopulateFunc_PopulateOnlySpecifiedGroups(t *testing
-		"metric_1":    3,
-		"metric_2":    4,
-		"entityName":  "playground:test:testEntity12-generated",
-		"displayName": "testEntity12-generated",
-		"clusterName": "playground",
-	}
-	expectedEntityData2.Metrics = []metric.MetricSet{expectedMetricSet2}
+	expectedEntityData2, err := dummyIntgr.Entity("testEntity12-generated", "playground:test")
+	require.NoError(t, err)
 
-	expectedEntityData3, err := sdk.NewEntityData("playground", "k8s:cluster")
-	if err != nil {
-		t.Fatal()
+	expectedMetricSet2 := &metric.Set{
+		Metrics: map[string]interface{}{
+			"event_type":  "TestSample",
+			"metric_1":    float64(3),
+			"metric_2":    float64(4),
+			"displayName": "testEntity12-generated",
+			"clusterName": "playground",
+		},
 	}
-	expectedMetricSet3 := metric.MetricSet{
-		"event_type":        "K8sClusterSample",
-		"entityName":        "k8s:cluster:playground",
-		"clusterName":       "playground",
-		"clusterK8sVersion": "v1.15.42",
+	expectedEntityData2.Metrics = []*metric.Set{expectedMetricSet2}
+
+	expectedEntityData3, err := dummyIntgr.Entity("playground", "k8s:cluster")
+	require.NoError(t, err)
+
+	expectedMetricSet3 := &metric.Set{
+		Metrics: map[string]interface{}{
+			"event_type":        "K8sClusterSample",
+			"clusterName":       "playground",
+			"clusterK8sVersion": "v1.15.42",
+		},
 	}
-	expectedEntityData3.Metrics = []metric.MetricSet{expectedMetricSet3}
-	expectedInventory := sdk.Inventory{}
-	expectedInventory.SetItem("cluster", "name", "playground")
-	expectedInventory.SetItem("cluster", "k8sVersion", "v1.15.42")
+	expectedEntityData3.Metrics = []*metric.Set{expectedMetricSet3}
+	expectedInventory := inventory.New()
+	_ = expectedInventory.SetItem("cluster", "name", "playground")
+	_ = expectedInventory.SetItem("cluster", "k8sVersion", "v1.15.42")
 	expectedEntityData3.Inventory = expectedInventory
 
-	integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", new(struct{}))
-	if err != nil {
-		t.Fatal()
-	}
-	populated, errs := IntegrationProtocol2PopulateFunc(
-		integration,
+	dummyIntgr.Clear()
+	populated, errs := IntegrationPopulator(
+		dummyIntgr,
 		defaultNS,
 		&version.Info{GitVersion: "v1.15.42"},
 		fromGroupMetricSetTypeGuessFunc,
-		metricsNamingManipulator,
-		clusterMetricsManipulator,
 	)(groups, withGeneratorSpec)
 	assert.True(t, populated)
 	assert.Empty(t, errs)
-	assert.Contains(t, integration.Data, &expectedEntityData1)
-	assert.Contains(t, integration.Data, &expectedEntityData2)
-	assert.Contains(t, integration.Data, &expectedEntityData3)
-	assert.Len(t, integration.Data, 3)
+	assert.Len(t, dummyIntgr.Entities, 3)
+
+	// Search for missing entities using cmp
+	compareIgnoreFields := cmpopts.IgnoreUnexported(integration.Entity{}, metric.Set{}, inventory.Inventory{})
+	for _, ti := range []*integration.Entity{
+		expectedEntityData1,
+		expectedEntityData2,
+		expectedEntityData3,
+	} {
+		found := false
+		closestMatchDiff := ""
+		for _, presentEntity := range dummyIntgr.Entities {
```

Comment:
```suggestion
		for _, testEntity := range dummyIntgr.Entities {
```

Reply:
I think that `testEntity` would make me think I'm iterating over the synthetic ones (used for test) 😅

Reply:
If we rename `dummyIntgr` to `intgr`, I'm fine with naming it just `entity` maybe.